### ISSUE TITLE
TE unexpectedly closes when a second add-on options button is clicked

### ISSUE THREAD

**Issue by ccfs** [tablacus/TablacusExplorer#525]
This problem has been happening in recent versions.
![20211231_161848](https://user-images.githubusercontent.com/34089121/147814502-ce3aa0e0-89ff-4f11-868a-39230b9cc2fe.png)
Choose Tools > Add-ons
![20211231_161922](https://user-images.githubusercontent.com/34089121/147814503-d8819da4-c03b-42ed-bffa-cd536943a753.png)
Click "Options" button of "Mouse"
![20211231_161942](https://user-images.githubusercontent.com/34089121/147814507-db13304b-b170-495a-add0-f318aa2a24be.png)
Click "Options" button of "Key"
![20211231_161949](https://user-images.githubusercontent.com/34089121/147814508-63b1798b-076d-4033-9546-351ba6ed0c00.png)
TE unexpectedly closes.

This problem happens only if "**Segoe Fluent Icons**" add-on is **disabled**.
![20211231_171001](https://user-images.githubusercontent.com/34089121/147814537-7e871e2a-3fa3-4ff6-9d86-b4761a61b16f.png)

**Comment by JoxCoswit** [comment]
Yes but this also happens with "initial folder" (init files) when it just **turn off** "_Sagoe Fluent Icons_" (Not Enable)
It can no longer be started and the application is constantly crashing when it wants to start.
As far as I've noticed I've tried earlier versions the same thing happens with - e.g. te211221, te211224 etc.

Regards, 
Jox.

**Comment by tablacus** [comment]
I have not encountered this yet.
Let me know more details.
https://github.com/tablacus/TablacusExplorer/wiki/report-problem

Regards,

**Comment by JoxCoswit** [comment]
**I made video as example for demonstration with "initial folder" (te211221.zip) and info of environment:**
_I hope it will help for you._

https://user-images.githubusercontent.com/60333254/147877794-aeded5d8-f89e-49f8-a90a-7c660462ebb4.mp4




TE64 21.12.21 Microsoft Windows 10 Pro 64-bit 20H2 (10.0.19042) IE/11 JS/11.0.16384 mouse 1.33,key 1.26,addressbar 1.69,filterbar 1.38,back 1.18,forward 1.18,up 1.20,treeview 1.43,toolbar 1.43,mainmenu 1.16,titlebar 1.09,tabplus 1.89,multithread 1.28,remember 1.40,undoclosetab 1.15,aboutblank 1.02,segoeicons 1.03

**Comment by ccfs** [comment]
May WebView2 involve in the problem?

**Comment by tablacus** [comment]
I have no problem with Windows 7, Windows 8.1 or Windows 10 in my environment.

Try the following 
1. Turn off "Sagoe Fluent Icons".
2. Get Icons...
3. Install "Tablacus Light".
4. Click OK.

![image](https://user-images.githubusercontent.com/5156977/147907710-c1e1a786-2855-4828-92c0-1ad81fcc7886.png)

Regards,

**Comment by ccfs** [comment]
The problem is not using any icon set. On the contrary, due to the unexpected closing, users are forced to enable "Segoe Fluent Icons" add-on and to use those icons, and cannot use the traditional image icons.

I think there is something wrong in TE in changing certain buttons (Portable, Browse, etc.) from texts to symbols in add-ons.

Regards,

**Comment by JoxCoswit** [comment]
The TE crash only when it is turn off "Sagoe Fluent Icons" when others ADD-ON are turned off everything is fine (by my test)
I did additional tests, and checked which version this problem comes from... these are the results:

_2021_
te211211 - **OK**
te211214 - **I can't find it, as if it doesn't exist** _(screenshot below)_
te211215 - **this problem begins...**
te211231 - **it still exists**
_2022_
te220101 - **the same thing** (Pre-release, latest version)

But then I came up with the idea to exclude everything in order and finally "Sagoe Fluent Icons"; an interesting thing happened -  there was no crashed. Everything works fine, if the rest ADD-ONS are all turn off, but in case - the next set is included in the combination (Turn ON) "Back"/"Forward"/"Up" - crashed start  ("Sagoe Fluent Icons" is turn off)
If same of them turn off but enable "Back"/"Forward" or "Forward"/"Up" or "Back"/"Up" - there no crashed
From the beginning "Sagoe Fluent Icons" is turn off  as well as the rest ADD-ONS.

I apologize in advance if I wrote something wrong but I did it the way I only knew how to do this test by the method of failures and attempts and in the end what I concluded from that, I’m not claiming that’s exactly the case, maybe I predicted something or made a mistake in testing set - these are only my results that I get at the end.

As a conclusion, in order for a crash to occur - all of these three (3) must be included together... or at least some of them
_For "Sagoe Fluent Icons" I would say he was just a trigger to start with crashed._
- Back 1.18
- Forward 1.18
- Up 1.20

**te211214**
![te211214](https://user-images.githubusercontent.com/60333254/147916500-817f707f-7d38-4062-a0e9-159292875e03.jpg)


Regards, 
Jox.

**Comment by JoxCoswit** [comment]
**Video as proof of upper testing**   

1. **2021-12-11** (working without any problems, if only "Sagoe Fluent Icons" is turned off)
2. **2022-01-01** (if "Sagoe Fluent Icons" is turned off, but turn enabled "Back"/"Forward"/"Up")

**2021-12-11** 

- working without any problems, if only "Sagoe Fluent Icons" is turned off: 

https://user-images.githubusercontent.com/60333254/147928677-1c791b22-fb21-4032-86a7-0ed82a158354.mp4


**2022-01-01** 

- if "Sagoe Fluent Icons" is turned off, but turn enabled "Back"/"Forward"/"Up"

https://user-images.githubusercontent.com/60333254/147928814-36daf1e6-8836-4609-b450-304541850265.mp4

Regards,
Jox.

**Comment by tablacus** [comment]
@JoxCoswit Thank you for the detailed information. 

How about 21.12.13?
https://github.com/tablacus/TablacusExplorer/releases/tag/21.12.13

Best regards,

**Comment by JoxCoswit** [comment]
Same as te211211 no problems. 

Regards,
Jox.

**Comment by tablacus** [comment]
This is [te211214.zip](https://github.com/tablacus/TablacusExplorer/files/7802241/te211214.zip)
21.12.14 has been canceled due to a problem with Blink version.
I hope you can try this as well.

Regards.

**Comment by JoxCoswit** [comment]
te211214.zip is working fine, there is no problems either.

Regards,
Jox.

**Comment by tablacus** [comment]
Would you also test the one with only the dll replaced with 21.12.15 on 21.12.14?

[te211214+1.zip](https://github.com/tablacus/TablacusExplorer/files/7802372/te211214%2B1.zip)

Regards,

**Comment by JoxCoswit** [comment]
Yes, it's working fine!
There is no crashing in this version whatsoever.

Regards,
Jox.

**Comment by tablacus** [comment]
I'm not confident, but I adjusted it in Tablacus Explorer 22.1.3.
Please give it try.

https://github.com/tablacus/TablacusExplorer/releases/tag/22.1.3

Regards,

**Comment by JoxCoswit** [comment]
I try the latest/newest version Tablacus Explorer (te220103 - 22.1.3.) on my environment, - ie. Windows 10.
It's OK. I didn't find any of no previous problems with crashing.

I would ask also a colleague @ccfs for confirmation... is it okay from his side!?
_I don't have/use anymore  Windows 7._ 

Regards,
Jox.

**Comment by tablacus** [comment]
@JoxCoswit Thank you for your cooperation.

**Comment by JoxCoswit** [comment]
_**A brief comment on my observation:**_

Afterwards I tried to make the same thing on my laptop which also has install Windows 10 - and I was surprised!
There none of that happened to me what I had on my main desktop PC computer it didn't apper on laptop. 
Everything that happened at PC, none of that happened to my LAPTOP. 

I tried to turn OFF/ON ADD-ONS but there has never-ever been a single crashing of the TE (te211215-te220101) 

It seems to me that there are really a lot of factors that can affect starting from the operating system, configuration (processor, graphics...), system updates, settings, configuration etc. 

It is not possible to expect in general that will work and behave everywhere equally regardless of the similar settings or a slightly different configuration _(the laptop has a slightly diffrent- also weaker configuration/without newest windows updates)_

I also tried the today latest version (te220103) on my laptop and there were no problems or any unknown behavior.


Regards,
Jox.
